File: _API_TESTING_GHERKIN_/DEMOAPP004_PYTHON_PLAYWRIGHT/screenplay/abilities/use_token_parsers.py
"""Ability exposing token parser utilities."""

from tokenparser.date_parser import parse_date_range_token, parse_date_token
from tokenparser.dynamic_string_parser import generate_dynamic_string


class UseTokenParsers:
    def parse_date(self, token: str):
        return parse_date_token(token)

    def parse_date_range(self, token: str):
        return parse_date_range_token(token)

    def parse_dynamic_string(self, token: str) -> str:
        return generate_dynamic_string(token)

File: _API_TESTING_GHERKIN_/DEMOAPP001_TYPESCRIPT_CYPRESS/screenplay/abilities/UseTokenParsers.ts
import type { Ability } from "../core/types";
import {
  TokenDateParser,
  TokenDynamicStringParser,
  DateRange,
} from "tokenparser-api-shared";

export class UseTokenParsers implements Ability {
  get description(): string {
    return "Use token parsing utilities";
  }

  parseDateToken(token: string): Date {
    return TokenDateParser.parseDateStringToken(token);
  }

  parseDateStringToken(token: string): Date {
    return TokenDateParser.parseDateStringToken(token);
  }

  parseDateRangeToken(token: string): DateRange {
    return TokenDateParser.parseDateStringToken_DateRange(token);
  }

  parseDynamicString(token: string): string {
    return TokenDynamicStringParser.parseAndGenerate(token);
  }

  parseTokenizedStringLines(token: string): string[] {
    const generated = this.parseDynamicString(token);
    return generated.split(/\r?\n/).filter((line) => line.length > 0);
  }
}
